src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java

```diff
@@ -2790,7 +2791,7 @@ final void doExecutePreparedStatementBatch(PrepStmtBatchExecCmd batchCommand) th
                     // that repreparation is necessary.
                     ++numBatchesPrepared;
                     needsPrepare = doPrepExec(tdsWriter, batchParam, hasNewTypeDefinitions, hasExistingTypeDefinitions);
-                    if (needsPrepare || numBatchesPrepared == numBatches) {
+                    if (reqStarted || needsPrepare || numBatchesPrepared == numBatches) {
```

Comment:
I think this solution is a roundabout solution to the problem (which might also cause other regressions). The reason why we are getting an assertion error is because in AE enabled cases, we're retrieving the parameter encryption metadata twice in the second iteration of execute batch. Not only is this second retrieval of parameter encryption metadata hurting performance, it had the side effect of making the driver think that the request had been completed (getting the parameter encryption metadata counts as a command, which sets the `requestComplete` (and other flags) to true, which causes assertion error when we get to the actual execution of the prepared statement). You can check this behavior in the image below from the SQL profiler:

![image](https://user-images.githubusercontent.com/30157681/85638513-edab3400-b63a-11ea-8652-0ad7d4b99840.png)

If you were to compare this image to the non-AE path or the solution I've provided below, you'll see that the driver will no longer retrieve the paramter metadata twice.

My suggested solution is to instead call `encryptionMetadataIsRetrieved = true;` in line 2744 of this file, so that we only retrieve the param metadata once, which I believe is the correct way to solve this problem.

Let me know once you've confirmed if the fix will work (and tested it internally as well), so I can make a jar out of this PR and see if it solves the NPE that the user is experiencing from issue #1360.